cpp/tests/streaming/test_read_parquet.cpp

```diff
@@ -0,0 +1,237 @@
+#include <rapidsmpf/streaming/core/node.hpp>
+#include <rapidsmpf/streaming/cudf/owning_wrapper.hpp>
+#include <rapidsmpf/streaming/cudf/parquet.hpp>
+#include <rapidsmpf/streaming/cudf/table_chunk.hpp>
+
+#include "base_streaming_fixture.hpp"
+
+
+using namespace rapidsmpf;
+using namespace rapidsmpf::streaming;
+
+class StreamingReadParquet : public BaseStreamingFixture {
+  protected:
+    void SetUp() override {
+        BaseStreamingFixture::SetUp();
+        constexpr int nfiles = 10;
+        constexpr int nrows = 10;
+
+        temp_dir = std::filesystem::temp_directory_path() / "rapidsmpf_read_parquet_test";
+
+        for (int i = 0; i < nfiles; ++i) {
+            std::ostringstream filename_stream;
+            filename_stream << std::setw(3) << std::setfill(' ') << i << ".pq";
+            std::filesystem::path filepath = temp_dir / filename_stream.str();
+            source_files.push_back(filepath.string());
+        }
+
+        if (GlobalEnvironment->comm_->rank() == 0) {
+            std::filesystem::create_directories(temp_dir);
+
+            int start = 0;
+            for (auto& file : source_files) {
+                auto values = std::ranges::iota_view(start, start + nrows);
+                cudf::test::fixed_width_column_wrapper<int32_t> col(
+                    values.begin(), values.end()
+                );
+
+                std::vector<std::unique_ptr<cudf::column>> columns;
+                columns.push_back(col.release());
+                auto table = std::make_unique<cudf::table>(std::move(columns));
+
+                cudf::io::sink_info sink{file};
+                auto options =
+                    cudf::io::parquet_writer_options::builder(sink, table->view())
+                        .build();
+                cudf::io::write_parquet(options);
+                start += nrows + nrows / 2;
+            }
+        }
+
+        GlobalEnvironment->barrier();
+    }
+
+    void TearDown() override {
+        GlobalEnvironment->barrier();
+
+        if (GlobalEnvironment->comm_->rank() == 0 && std::filesystem::exists(temp_dir)) {
+            std::filesystem::remove_all(temp_dir);
+        }
+
+        BaseStreamingFixture::TearDown();
+    }
+
+    [[nodiscard]] cudf::io::source_info get_source_info() const {
+        return cudf::io::source_info(source_files);
+    }
+
+    std::filesystem::path temp_dir;
+    std::vector<std::string> source_files;
+};
+
+class StreamingReadParquetThrottle : public StreamingReadParquet,
+                                     public ::testing::WithParamInterface<int> {};
+
+INSTANTIATE_TEST_SUITE_P(
+    InvalidMaxTickets, StreamingReadParquetThrottle, ::testing::Values(-1, 0)
+);
+
+TEST_P(StreamingReadParquetThrottle, NonPositiveThrottleThrows) {
+    if (GlobalEnvironment->comm_->nranks() > 1) {
+        // Test is independent of size of communicator, so don't bother if we
+        // have more than one rank.
+        GTEST_SKIP();
+    }
```

Comment:
For convenience, it might be helpful to also run it in a multi-rank setup.
```suggestion
    if (GlobalEnvironment->comm_->rank() != 0) {
        // Test is independent of size of communicator.
        GTEST_SKIP() << "Test only runs on rank 0";
    }
```